packages/edit-site/src/components/editor/index.js

```diff
@@ -152,12 +152,41 @@ export default function Editor() {
 	// action in <URlQueryController> from double-announcing.
 	useTitle( hasLoadedPost && title );
 
-	if ( ! hasLoadedPost ) {
-		return <CanvasSpinner />;
-	}
+	const { hasResolvingSelectors } = useSelect( ( select ) => {
+		return {
+			hasResolvingSelectors: select( coreStore ).hasResolvingSelectors(),
+		};
+	} );
+	const [ loaded, setLoaded ] = useState( false );
+	const timeoutRef = useRef( null );
+
+	useEffect( () => {
+		if ( ! hasResolvingSelectors && ! loaded ) {
+			clearTimeout( timeoutRef.current );
+
+			/*
+			 * We're using an arbitrary 1s timeout here to catch brief moments
+			 * without any resolving selectors that would result in displaying
+			 * brief flickers of loading state and loaded state.
+			 *
+			 * It's worth experimenting with different values, since this also
+			 * adds 1s of artificial delay after loading has finished.
+			 */
+			timeoutRef.current = setTimeout( () => {
+				setLoaded( true );
+			}, 1000 );
+
+			return () => {
+				clearTimeout( timeoutRef.current );
+			};
+		}
+	}, [ loaded, hasResolvingSelectors ] );
+
+	const isLoading = ! loaded || ! hasLoadedPost;
```

Comment:
Potentially this logic to compute the isLoading const could be extracted to a custom hook `useSiteEditorLoading` or something.

Reply:
While I don't see the immediate need for that, I'm happy to go ahead and extract it when it can actually be reused somewhere.

Reply:
I think this is a good idea, I'll follow up in another PR.

Reply:
Addressed in #50511